power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -714,52 +763,177 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
   public:
     template <class Component> using UpdateType = typename Component::UpdateType;
 
-    static bool is_update_independent(ConstDataset const& update_data) {
-        // If the batch size is (0 or) 1, then the update data for this component is 'independent'
-        if (update_data.batch_size() <= 1) {
-            return true;
-        }
+    UpdateCompIndependence check_components_independence(ConstDataset const& update_data) const {
+        auto const all_comp_count_in_base = this->all_component_count();
+        Idx const n_scenarios = update_data.batch_size();
 
-        auto const process_buffer_span = []<typename CT>(auto const& all_spans) -> bool {
+        auto check_ids_na = [](auto const& all_spans) {
+            std::vector<std::vector<bool>> ids_na{};
+            for (const auto& span : all_spans) {
+                std::vector<bool> id_na{};
+                if constexpr (requires { span.front().id; }) {
+                    for (const auto& obj : span) {
```

Comment:
cfr. conventions in rest of repo
```suggestion
            for (auto const& span : all_spans) {
                std::vector<bool> id_na{};
                if constexpr (requires { span.front().id; }) {
                    for (auto const& obj : span) {
```

Reply:
Addressed.